numba/np/arrayobj.py

```diff
@@ -1957,17 +1957,23 @@ def impl(start, stop, num=50):
                 raise ValueError('Geometric sequence cannot include zero')
             start = result_dtype(start)
             stop = result_dtype(stop)
-            both_imaginary = (start.real == 0) & (stop.real == 0)
-            both_negative = (np.sign(start) == -1) & (np.sign(stop) == -1)
-            out_sign = 1
-            if both_imaginary:
-                start = start.imag
-                stop = stop.imag
-                out_sign = 1j
-            if both_negative:
-                start = -start
-                stop = -stop
-                out_sign = -out_sign
+            if numpy_version < (2, 0):
```

Comment:
Need to find out why `np.geomspace` testing only started to fail against NumPy 2.1, the code change was present in NumPy 2.0 (redefinition of complex `sign`)?